test/round_trip_test.dart

```diff
@@ -81,4 +87,22 @@ main() async {
     expect(await testClient.stream(1).toList(), [1, 2, 3]);
     server.shutdown();
   });
+
+  test('exception in onMetadataException', () async {
+    final Server server = Server([TestServiceWithOnMetadataException()]);
+    await server.serve(address: 'localhost', port: 0);
+
+    final channel = FixedConnectionClientChannel(Http2ClientConnection(
+      'localhost',
+      server.port,
+      ChannelOptions(credentials: ChannelCredentials.insecure()),
+    ));
+    final testClient = TestClient(channel);
+    await testClient
```

Comment:
Consider

```dart
await expectLater(testClient.stream(1).toList(), throwsA(isA<GrpcError>()));
```

Because the way the test is currently written does not actually verify that the error is being thrown, it just check that if it is thrown then it would be `GrpcError`.